### ISSUE TITLE
Blazor WASM: Publish from MSBuild does not create wwwroot\_framework folder

### ISSUE THREAD

**Issue by Gambero81** [dotnet/aspnetcore#29626]
I use Visual studio 2019 v16.8.4

Create a new Blazor WASM app, and publish.

**If i publish from visual studio all work well.**

**if i publish with MSBuild.exe, the publish wwwroot subfolder does not include the _framework folder** and running the app fail with error 404.

i use this msbuild command:
`"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe" "%SolutionFilePath%" /t:Rebuild /p:Configuration=Release /p:DeployOnBuild=true /p:PublishProfile=Release /p:DebugSymbols=false /p:DebugType=None /p:AllowedReferenceRelatedFileExtensions=none /fl`

in attachment a sample project to reply the issue,

What's wrong? i miss a msbuild parameter?
Thanks

[BlazorApp18.zip](https://github.com/dotnet/aspnetcore/files/5874206/BlazorApp18.zip)

**Comment by javiercn** [comment]
@Gambero81 It's a bit strange that you are using full framework MSBuild to publish the app from the command-line.

Is there a reason why `dotnet publish` doesn't work for you? That's the recommended way to publish from the command-line and what we test for.

**Comment by Gambero81** [comment]
@javiercn the problem seem to be related to recent upgrade of my project to net5.

Publish with MSBuild with .net core 3.1 work well.  

I will try doing it with dotnet publish and reply asap.

**Comment by pranavkm** [comment]
@Gambero81 could you use the `Publish` target instead of using `/p:DeployOnBuild`. Also I only tried publishing the Server project rather than the sln:

```
set SolutionFilePath=%ProjectPath%\BlazorApp18\Server\BlazorApp18.Server.csproj

set MSBuild=C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\MSBuild.exe
set MSBuildParameters="%SolutionFilePath%" /restore /t:Publish /p:Configuration=Release /v:M /p:PublishProfile=Release /p:DebugSymbols=false /p:DebugType=None /p:AllowedReferenceRelatedFileExtensions=none
```

It looks you get different sets of targets invoked when using `/p:DeployOnBuild=true` vs `/t:Publish`. Probably something to look in to.

**Comment by cm-andrews** [comment]
I am also seeing some weirdness here. The _framework folder is missing when doing a simple `dotnet publish -c Release` from within the VS Developer Powershell or on an Ubuntu Server with the 5.0.2 SDK installed. Running the same command from regular Powershell on Windows creates the _framework folder as expected. All concerning Blazor WASM on 5.0.2. Maybe related to #21178?

**Comment by tai-yi** [comment]
Tried many times, details show in below, any idea?

* `publish` in solution folder, there is no `_framework` published in BlazorApp.Server output folder.
* `publish` in BlazorApp.Server folder, `_framework` is published in BlazorApp.Server output folder.

```
app.sln
- BlazorApp
  BlazorApp.Client
  BlazorApp.Server
-ApiApp2
 ApiApp2.csproj
```

**Comment by carios23** [comment]
We are having this issue. It has been 6 months already. Does anybody has a workaround for this? We are having to publish directly from Visual Studio.

**Comment by Gambero81** [comment]
Hi @carios23 i solved using `/p:Publish` instead of `/p:DeployOnBuild`

**Comment by carios23** [comment]
> 
> 
> Hi @carios23 i solved using `/p:Publish` instead of `/p:DeployOnBuild`

I replaced with /t:Publish, but now the following task for unit tests is failing

- task: VSTest@2
  inputs:
    testSelector: 'testAssemblies'
    codeCoverageEnabled: true
    testAssemblyVer2: |
      **\*.Tests.dll
      !*testhost.dll
      !**\obj\**
      !**\bin\**\ref\**
      !**\xunit.runner.visualstudio.testadapter.dll
      !**\xunit.runner.visualstudio.dotnetcore.testadapter.dll
    searchFolder: '$(System.DefaultWorkingDirectory)'
    otherConsoleOptions: '/Framework:.NETCoreApp,Version=5.0'

**Comment by Pilchie** [comment]
@javiercn - should we go ahead and close this?

**Comment by javiercn** [comment]
@Pilchie Yep, I recently confirmed after reenabling a test that was skipped for the wrong reasons. This is fix in 6.0
